apps/site/src/app/login/components/LoginForm.tsx

```diff
@@ -1,31 +1,61 @@
-import ValidatingForm from "@/lib/components/ValidatingForm/ValidatingForm";
+"use client";
 
-const EMAIL_REGEX = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;
+import { FormEvent, useState } from "react";
+
+import RequiredAsterisk from "@/app/apply/sections/Components/RequiredAsterisk";
+import Button from "@/lib/components/Button/Button";
+
+const EMAIL_REGEX = /^\w+([\.\-]?\w+)*@\w+([\.\-]?\w+)*(\.\w{2,3})+$/;
 const LOGIN_PATH = "/api/user/login";
 
 function LoginForm() {
+	const [validated, setValidated] = useState<boolean | null>(null);
+
+	const handleSubmit = (event: FormEvent<HTMLFormElement>): void => {
+		const form = event.currentTarget;
+		if (!form.checkValidity()) {
+			// prevent submission to display validation feedback
+			event.preventDefault();
+			setValidated(false);
+		} else {
+			setValidated(true);
+		}
+	};
```

Comment:
Suggestion: the `ValidatingForm` was designed to be a shared component between `LoginForm` and the guest `VerificationForm` since both will want to use custom form validation. Could we adapt `ValidatingForm` to fit both components?

Reply:
Sure, but how would we pass the validated state down to the children?

Reply:
Ahh, I'm a bit too used to React Bootstrap. Looks like the way they do it is by applying a `was-validated` class to the form element based on the prop ([Form.tsx:56](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Form.tsx#L56)) and then Bootstrap displays the validation accordingly.
```css
.was-validated :invalid ~ .invalid-feedback { display: block }
.invalid-feedback { display: none }
```
We don't necessarily have to do it the same way, but this seems rather straightforward. Otherwise, there might be some way with `useContext` but that seems a bit complicated.

Reply:
Implemented in `[5f33e63](https://github.com/HackAtUCI/irvinehacks-site-2024/pull/114/commits/5f33e636c09e5c74ac95d72aaf576342ff434bbf)`